docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+| Status | Authors   | Created    | 
+|--------|-----------|------------|
+| Review | @jewertow | Jan 2 2025 |
+
+# CRDs and controllers
+
+## Overview
+
+Currently, we use helm to manage the controller deployment and its configuration. The configuration is passed to the controller as CLI arguments.
+This approach requires restarting the controller on any configuration change and causes pod failures in case of a misconfiguration.
```

Comment:
```suggestion
Currently, we use Helm to manage the controller deployment and its configuration, which is passed as CLI arguments. This approach offers limited flexibility, as any configuration change requires a controller restart and can lead to pod failures in case of misconfiguration.
```